### PR TITLE
Add example for specifying figure size in different units

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

This explains and illustrates best practices for working with figure sizes in units differing from the default inches.

This addresses the user needs of #12402 as good as possible without introducing new API.

### CODE REVIEW

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+plt.subplots(figsize=(15*cm, 5*cm))
+plt.text(0.5, 0.5, '15cm x 5cm', **text_kwargs)
+plt.show()
+
+
+#############################################################################
+# Figure size in pixel
+# --------------------
+# Similarly, one can use a conversion from pixels.
+#
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/figure_size_units.py:43]: I think you should remind people that the dpi can change when saving the figure....

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+#############################################################################
+# Figure size in cm
+# -----------------
+# Multiplying cm-based numbers with a conversion factor from cm to inches,
+# gives the right numbers. Naming the conversion factor ``cm`` makes the
+# conversion almost look like appending a unit to the number, which is nicely
+# readable.
+#
+cm = 1/2.54  # cm in inches
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/figure_size_units.py:33]: I was 90% sure this was exact, but had to look it up to be 100% sure.  You may want to indicate that with a link to https://en.wikipedia.org/wiki/Inch? and/or more decimal places?  Or at least say its exact?

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# Because of the default `rcParams['figure.dpi'] = 100` [*]_, one can mentally
+# divide the needed pixel value by 100:
+#
+plt.subplots(figsize=(6, 2))
+plt.text(0.5, 0.5, '600px x 200px', **text_kwargs)
+plt.show()
+
+#############################################################################
+# .. [*] The ``matplotlib inline`` backend in jupyter currently still uses a
+#        different default of `rcParams['figure.dpi'] = 72`.
```
Comment (tacaswell) [inline examples/subplots_axes_and_figures/figure_size_units.py:63]: This should also have a caveat that inline use `bbox_inches='tight'` so the actually size in pixels is unpredictable.

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+# References
+# """"""""""
+#
+# The use of the following functions and methods is shown in this example:
+
+import matplotlib
+
+matplotlib.pyplot.figure
+matplotlib.pyplot.subplots
+matplotlib.pyplot.subplot_mosaic
```
[the remark this answers is not in the document]
Comment (tacaswell) [reply examples/subplots_axes_and_figures/figure_size_units.py:78]: This is to trigger the _back_ links which I think is useful here (as these are the entry points where we want people to use to set the figure size)?

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
+"""
+==============================
+Figure size in different units
+==============================
+
+The native figure size unit in matplotlib is inches, deriving from print
```
Comment (QuLogic) [inline examples/subplots_axes_and_figures/figure_size_units.py:6]:
```suggestion
The native figure size unit in Matplotlib is inches, deriving from print
```

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+industry standards. However, users may need to specify their figures in other
+units like cm or pixels. This example illustrates how to do this efficiently.
```
Comment (QuLogic) [inline examples/subplots_axes_and_figures/figure_size_units.py:8]: You should spell out centimetres, just like inches, though using `cm` for the scale factor is fine.
```suggestion
units like centimetres or pixels. This example illustrates how to do this efficiently.
```

File: examples/subplots_axes_and_figures/figure_size_units.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+#
+plt.subplots(figsize=(6, 2))
+plt.text(0.5, 0.5, '6 inches x 2 inches', **text_kwargs)
+plt.show()
+
+
+#############################################################################
+# Figure size in cm
+# -----------------
+# Multiplying cm-based numbers with a conversion factor from cm to inches,
```
Comment (QuLogic) [inline examples/subplots_axes_and_figures/figure_size_units.py:28]:
```suggestion
# Figure size in centimetre
# -------------------------
# Multiplying centimetre-based numbers with a conversion factor from centimetre to inches,
```

File: examples/subplots_axes_and_figures/figure_size_units.py (hunk 2 above)
Comment (QuLogic) [inline examples/subplots_axes_and_figures/figure_size_units.py:33]:
```suggestion
cm = 1/2.54  # centimetres in inches
```

Comment (timhoffm) [reply examples/subplots_axes_and_figures/figure_size_units.py]: IMHO that'd be overly cautious. One should assume that code/logic is correct unless explicitly indicated as an approximation. I'd really feel awkward adding something like "Look, what we're doing is really correct. Here is the proof".

Comment (jklymak) [reply examples/subplots_axes_and_figures/figure_size_units.py]: OK, but doing `1/2.5400` wouldn't hurt?

**Pull request merged**